// @magidoc/plugin-svelte-marked/markdown/markedConfiguration.js
import { MarkdownHeading, MarkdownBloquote, MarkdownList, MarkdownListItem, MarkdownBr, MarkdownCode, MarkdownCodeSpan, MarkdownTable, MarkdownHtml, MarkdownParagraph, MarkdownLink, MarkdownText, MarkdownDfn, MarkdownDel, MarkdownEm, MarkdownHr, MarkdownStrong, MarkdownImage, MarkdownSpace, } from './components';
import { marked, Slugger } from 'marked';
export function parse(src) {
    const lexer = new marked.Lexer();
    return lexer.lex(src);
}
export const defaultRenderers = () => ({
    heading: MarkdownHeading,
    blockquote: MarkdownBloquote,
    list: MarkdownList,
    list_item: MarkdownListItem,
    br: MarkdownBr,
    code: MarkdownCode,
    codespan: MarkdownCodeSpan,
    table: MarkdownTable,
    html: MarkdownHtml,
    paragraph: MarkdownParagraph,
    link: MarkdownLink,
    text: MarkdownText,
    def: MarkdownDfn,
    del: MarkdownDel,
    em: MarkdownEm,
    hr: MarkdownHr,
    strong: MarkdownStrong,
    image: MarkdownImage,
    space: MarkdownSpace,
    escape: MarkdownSpace,
});
export const defaultOptions = () => ({
    baseUrl: '/',
    slugger: new Slugger(),
});
